src/main/java/gregtech/api/metatileentity/MetaTileEntity.java

```diff
@@ -894,10 +896,7 @@ public void receiveInitialSyncData(PacketBuffer buf) {
         int amountOfTraits = buf.readShort();
         for (int i = 0; i < amountOfTraits; i++) {
             int traitNetworkId = buf.readVarInt();
-            mteTraits.stream()
-                    .filter(otherTrait -> otherTrait.getNetworkID() == traitNetworkId)
-                    .findAny()
-                    .ifPresent(trait -> trait.receiveInitialData(buf));
+            Objects.requireNonNull(mteTraitByNetworkId.get(traitNetworkId)).receiveInitialData(buf);
```

Comment:
Hm, this is slightly different behavior. The new behavior will crash if passed an invalid network id, while the old behavior would have just not received any data. It might be better to do

```java
MTETrait trait = mteTraitByNetworkId.get(traitNetworkId);
if (trait != null) {
        trait.recieveInitialData(buf);
}
```

And then log a warning if an invalid trait was received.